semantic/monotype.go

```diff
@@ -442,20 +425,226 @@ func (l MonoType) Equal(r MonoType) bool {
 	return false
 }
 
+func newBasicType(t fbsemantic.Type) MonoType {
+	builder := flatbuffers.NewBuilder(16)
+	offset := buildBasicType(builder, t)
+	builder.Finish(offset)
+
+	buf := builder.FinishedBytes()
+	basic := fbsemantic.GetRootAsBasic(buf, 0)
+	mt, err := NewMonoType(basic.Table(), fbsemantic.MonoTypeBasic)
+	if err != nil {
+		panic(err)
+	}
+	return mt
+}
+
 func NewArrayType(elemType MonoType) MonoType {
-	//builder := flatbuffers.NewBuilder(1024)
-	//fbsemantic.ArrStart(builder)
-	//fbsemantic.AddTType(builder, elemType.Type())
-	//// TODO (algow): how do we inject the elemType?
-	//fbsemantic.AddT(builder, flatbuffer.UOffsetT)
+	builder := flatbuffers.NewBuilder(32)
+	offset := buildArrayType(builder, elemType)
+	builder.Finish(offset)
+
+	buf := builder.FinishedBytes()
+	arr := fbsemantic.GetRootAsArr(buf, 0)
+	mt, err := NewMonoType(arr.Table(), fbsemantic.MonoTypeArr)
+	if err != nil {
+		panic(err)
+	}
+	return mt
+}
+
+type ArgumentType struct {
+	Name     []byte
```

Comment:
Should this be a string to make it easier for callers of these methods? I don't have a strong preference

Reply:
I chose a byte since that's the API for what gets returned by flatbuffers and using a string would incur a memory allocation. I figured that it was more performant to go from `string -> []byte` than `[]byte -> string`.

Reply:
Works for me